### PR TITLE
gh #268 CompositeIn L3 testcase for new API

### PR DESCRIPTION
**Pull request by thanushreevani**
Added L3 test case
Updated the spec and test procedure doc

### CODE REVIEW

File: host/tests/L3_TestCases/dsCompositeIn/dsCompositeIn_test5_VerifyVideoMode_Callback.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+        Args:
+            port_type (str) : CompositeIn port
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True and connect == True:
```
Comment (srinivasgtl) [inline host/tests/L3_TestCases/dsCompositeIn/dsCompositeIn_test5_VerifyVideoMode_Callback.py:67]: Can we improve the code readability here?
 if manual:
        message = (
            f"Check if CompositeIn source is connected to {port_type} and press Y:"
            if connect
            else f"Change the resolution of the video played from CompositeIn source device connected to {port_type} and press Y:"
        )
        return self.testUserResponse.getUserYN(message)

    # TODO: Add automated verification methods here
    return False

**Review by srinivasgtl — COMMENTED** [review]
suggestion to improve the code readability.

File: docs/pages/dsCompositeIn/ds-compositeIn-L3-Low-Level-TestSpec.md [hunk 2]
```diff
@@ -43,6 +44,7 @@ Below are top test use-case for the Composite Input.
 |2|Verify the CompositeIn active port status with callbacks|Connect a CompositeIn source device, select the CompositeIn port and check the callbacks is triggered when the active status changes(i.e like isPresented, activeport)|`dsCompositeInRegisterStatusChangeCB()`|
 |3|Scale the video and verify |Play the video stream(`vts_SDR_stream`) in CompositeIn source device connected to the active CompositeIn port and scale the video resolution|`dsCompositeInScaleVideo()`|
 |4|Verify the CompositeIn Signal change with callback| Connect a CompositeIn source device, select the CompositeIn port and check the callback is triggered when the change in signal status occurs(i.e like no signal , unstable signal, stable signal)|`dsCompositeInRegisterSignalChangeCB()`|
+|5|Verify the CompositeIn video mode change with callback| Connect a CompositeIn source device, select the CompositeIn port and check the callback is triggered when the change in video mode on source device occurs|`dsCompositeInRegisterVideoModeUpdateCB()`|
```
Comment (Anbukannadhasan) [inline docs/pages/dsCompositeIn/ds-compositeIn-L3-Low-Level-TestSpec.md:47]: Description can  be rephrase like this "Connect a CompositeIn source device, select the CompositeIn port, and verify that the callback is triggered when there is a change in the video mode on the source device."

File: host/tests/dsClasses/dsCompositeIn.py [hunk 3]
```diff
@@ -203,6 +203,34 @@ def getPortCallbackStatus(self):
...
+        Returns:
+            tuple:
+                - port (str): The CompositeIn port number as a string.
+                - videoResolution (str): Video resolution as a string.
+                - videoFramerate (str): Video framerate as a string.
+            None: If no matching signal status is found.
+        """
+
+        result = self.testSession.read_until("Received VideomodeChange callback port:")
+        callpattern = r"Received VideomodeChange callback port: \[(\w+)\], videoResolution: \[(\w+)\], videoFrameRate: \[(\w+)\]"
```
Comment (Anbukannadhasan) [inline host/tests/dsClasses/dsCompositeIn.py:226]: whether the string has been validated?

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: docs/pages/dsCompositeIn/ds-compositeIn-L3-Low-Level-TestSpec.md (hunk 2 above)
Comment (thanushreevani) [reply docs/pages/dsCompositeIn/ds-compositeIn-L3-Low-Level-TestSpec.md]: done

File: host/tests/dsClasses/dsCompositeIn.py (hunk 3 above)
Comment (thanushreevani) [reply host/tests/dsClasses/dsCompositeIn.py:226]: yes validated

**Review by srinivasgtl — APPROVED** [review]
ship it.

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Pull request merged**